src/java/org/apache/cassandra/io/util/FileUtils.java

```diff
@@ -177,15 +178,24 @@ public static File createDeletableTempFile(String prefix, String suffix)
 
     public static Throwable deleteWithConfirm(String filePath, boolean expect, Throwable accumulate)
     {
-        return deleteWithConfirm(new File(filePath), expect, accumulate);
+        return deleteWithConfirm(new File(filePath), expect, accumulate, null);
     }
 
     public static Throwable deleteWithConfirm(File file, boolean expect, Throwable accumulate)
     {
+        return deleteWithConfirm(file, expect, accumulate, null);
+    }
+    
+    public static Throwable deleteWithConfirm(File file, boolean expect, Throwable accumulate, RateLimiter rateLimiter)
+    {
+        if (rateLimiter == null)
```

Comment:
Instead of creating a new RateLimiter for each file deleted, we can guard the rateLimiter.acquire with the null check.

Reply:
(more for non-clear snapshots that call `deleteWithConfirm`)